test(latency during upgrade): based on other latency test

fgelcer opened the pull request:
used as base the acceptance criteria latency test,
the latency during operations, with 650GB dataset,
this test will run as the "operations" a rolling upgrade
on all the nodes.
currently it has the default of the base test (3 nodes)
on a single DC and same region. (more to be added).

QA-Task is https://github.com/scylladb/qa-tasks/issues/1191


## PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

fgelcer commented:
giving it a ride here --> https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/latency-during-upgrade/job/latency_650gb_during_upgrade/3/

the parameters are being sent manually through the pipeline, and if it starts correctly, i will start patching the pipeline itself to calculate the base version correctly

fgelcer commented:
seems like the test is working:
```
22:42:34  < t:2023-05-30 19:42:34,181 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-05-30 19:42:34.179: (InfoEvent Severity.NORMAL) period_type=not-set event_id=86bb8975-4616-411d-909c-c3fa193480be: message=Upgrade Node perf-latency-nemesis-ubuntu-db-node-0598bc82-2 begin
22:42:36  < t:2023-05-30 19:42:36,003 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-05-30 19:42:36.001: (InfoEvent Severity.NORMAL) period_type=not-set event_id=0c7caafe-fb16-4669-8bc3-3c13a076b4fd: message=Upgrading a Node
22:43:18  < t:2023-05-30 19:43:12,344 f:common.py       l:166  c:utils                p:INFO  > Verifying Scylla repo file
22:43:31  < t:2023-05-30 19:43:31,029 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-05-30 19:43:31.028: (NodetoolEvent Severity.NORMAL) period_type=begin event_id=b722dc18-bc5c-45b6-a290-87f2295b8745: nodetool_command=drain node=perf-latency-nemesis-ubuntu-db-node-0598bc82-2
22:43:56  < t:2023-05-30 19:43:54,050 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-05-30 19:43:43.809: (NodetoolEvent Severity.NORMAL) period_type=end event_id=b722dc18-bc5c-45b6-a290-87f2295b8745 duration=12s: nodetool_command=drain node=perf-latency-nemesis-ubuntu-db-node-0598bc82-2
22:43:56  < t:2023-05-30 19:43:54,054 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-05-30 19:43:43.810: (NodetoolEvent Severity.NORMAL) period_type=begin event_id=a266432c-bd31-40a6-8d1a-393d97f58524: nodetool_command=snapshot node=perf-latency-nemesis-ubuntu-db-node-0598bc82-2
22:43:57  < t:2023-05-30 19:43:57,824 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-05-30 19:43:45.576: (NodetoolEvent Severity.NORMAL) period_type=end event_id=a266432c-bd31-40a6-8d1a-393d97f58524 duration=1s: nodetool_command=snapshot node=perf-latency-nemesis-ubuntu-db-node-0598bc82-2
22:43:57  < t:2023-05-30 19:43:57,825 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-05-30 19:43:46.446: (JMXServiceEvent Severity.NORMAL) period_type=end event_id=364e475b-5aef-4613-b5d6-859970e92b96 duration=1h9m47s node=perf-latency-nemesis-ubuntu-db-node-0598bc82-2
22:43:57  < t:2023-05-30 19:43:57,826 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-05-30 19:43:46.448: (ScyllaServerStatusEvent Severity.NORMAL) period_type=end event_id=ab1d1675-a8ab-44c7-87a1-711fe39581e6 duration=1h9m51s node=perf-latency-nemesis-ubuntu-db-node-0598bc82-2
```

now, once we see the whole report stage working, i will return the test to the original dataset size, so we can get 1st time real results

fgelcer commented:
now that the PR is ready, and @aleksbykov helped me with the indexes in ES (on my previous run i have a failure on max number of fields in the index), i will run now all the 3 workloads, with full dataset:

https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/latency-during-upgrade/job/latency_650gb_during_upgrade/9/

fgelcer commented:
locally, all checks passed

fgelcer commented:
read run returned results [from job](https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/latency-during-upgrade/job/latency_650gb_during_upgrade/14/), but write failed with many error events, such as `com.datastax.driver.core.exceptions.OverloadedException: Queried host (ip-10-4-0-18.eu-west-1.compute.internal/10.4.0.18:9042) was overloaded: Too many in flight hints: 10486950`

i'm checking why we failed to send the email, as the results were collected and correctly calculated

temichus reviewed (commented):
Could you please also attach a new email report  for review?

fgelcer commented on vars/perfRegressionParallelPipeline.groovy:
```diff
@@ -32,6 +32,7 @@ def call(Map pipelineParams) {
 
             string(defaultValue: '', description: '', name: 'scylla_ami_id')
             string(defaultValue: '', description: '', name: 'scylla_version')
+            string(defaultValue: '', description: '', name: 'new_scylla_repo')
```
used here --> https://github.com/scylladb/scylla-cluster-tests/blob/e38ad755c3db62ee7ee9c6b67a5cd34936f6e515/upgrade_test.py#L150

fgelcer commented on performance_regression_test.py:
```diff
@@ -270,6 +271,18 @@ def _run_cql_commands(self, cmds, node=None):
                 session.execute(cmd)
 
+    def get_email_data(self):  # pylint: disable=no-self-use
+        """prepare data to generate and send via email
```
already moved, in latest pushes

temichus replied:
but it is still there in PerformanceRegressionTest

fgelcer replied:
moved

fgelcer commented:
> Could you please also attach a new email report for review?

i will, once i finish adding the base and target versions (almost done)

fgelcer commented:
ping @roydahan @temichus @aleksbykov

fgelcer commented:
@roydahan @temichus @aleksbykov , i forwarded you the 2 email reports i have generated [from my staging job](https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/latency-during-upgrade/job/latency_650gb_during_upgrade/) (builds 18 and 19)... this PR is ready for review... if we can do it today, i may be able to add it to the weekly triggers

fgelcer commented:
see attached report example:

![Screenshot from 2023-06-25 12-08-42](https://github.com/scylladb/scylla-cluster-tests/assets/32832995/2e773836-a938-4500-8c09-a4ac240d7cf4)

roydahan commented on performance_regression_test.py:
```diff
@@ -790,3 +791,102 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
+        self.preload_data()
+        self.wait_no_compactions_running()
+        self.run_fstrim_on_all_db_nodes()
+        self.fill_db_data_for_truncate_test(insert_rows=10)
```
Why do we need this?
Please explain in a comment.

fgelcer replied:
added

roydahan replied:
Please open a task for such flag (which seems pretty simple to me).
and mention in the task to remove these lines when adding it.

roydahan replied:
The "truncate during upgrade" is set as decorator during the upgrade node.
Having the truncate in this test is wrong!
The correct solution is to move the upgrade function to a shared library and use it here and in the upgrade_test where it has the decorator.

roydahan commented on performance_regression_test.py:
```diff
@@ -790,3 +791,102 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
+    @latency_calculator_decorator(legend="Upgrade Node")
+    def _upgrade_node(self, node):
+        InfoEvent(message='Upgrade Node %s begin' % node.name).publish()
+        self.upgrade_node(node)
```
Please add inside "upgrade_node" info events that will help to breakdown the operation smaller chunks.
Including:
upgrade_system
drain
snapshot
shutdown
packages installations
sysconfig_setup
start node
upgrade sstables

fgelcer replied:
added

fgelcer commented:
after last changes proposed by @roydahan , i'm running a full job run [here](https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/latency-during-upgrade/job/latency_650gb_during_upgrade/20/)

fgelcer commented:
ping @roydahan @temichus @aleksbykov review. PR is ready to be merged

roydahan commented:
@fgelcer please set the official jobs to run it in master and for release.
How do we  set the base versions per job?

fgelcer commented:
> @fgelcer please set the official jobs to run it in master and for release. How do we set the base versions per job?

right now, the way to run it is setting per build a "version" (it will be the base version) or an AMI, and we need to pass the `new_repo` URL, that will be the target version...
for now, until i can make the calculation of the base version from the `new_repo` URL, we can pass that value in the weekly trigger, or set a default in the pipeline itself

fgelcer commented:
creating the jobs, and will update the triggers once this PR will be merged:
https://jenkins.scylladb.com/view/master/job/scylla-master/job/scylla-master-perf-regression-latency-650gb-during-upgrade/
https://jenkins.scylladb.com/job/Perf-Regression/job/scylla-release-perf-regression-latency-650gb-during-upgrade/

aleksbykov commented on performance_regression_test.py:
```diff
@@ -790,3 +791,114 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
+            base_version, base_build_id = _get_version_and_build_id_from_node(node)
+            self._upgrade_node(node)
+            target_version, target_build_id = _get_version_and_build_id_from_node(node)
+            versions_list.append({'base_version': base_version,
+                                  'base_build_id': base_build_id,
+                                  'target_version': target_version,
+                                  'target_build_id': target_build_id
+                                  })
```
i suggest here to add also node name, so in any failure it will easy to see which node were upgraded or not

fgelcer replied:
@aleksbykov , added

aleksbykov reviewed (dismissed):
LGTM, small suggestion

fgelcer commented:
> There seems to be an issue with "nodetool upgradesstables" taking too long (but with the new events) it takes few seconds to spot it...
> 
> Please add a way to control if to run "upgradesstables" or not. For now, set it to "false".

perhaps we won't have to do it anymore, as it may be fixed by #6335

fgelcer commented:
running again [this job](https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/latency-during-upgrade/job/latency_650gb_during_upgrade/22/), since there was a fix to `upgradesstables` in SCT, so it should take much less time to run the operations now.

roydahan commented:
failed

fgelcer commented:
> failed

forgot to rebase the branch in my fork with latest master changes... rerunning [here](https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/latency-during-upgrade/job/latency_650gb_during_upgrade/23/)

fgelcer commented:
@roydahan , the whole upgrade of the 1st node took 7 minutes and a half:
```
2023-07-10 19:01:32.189: (InfoEvent Severity.NORMAL) period_type=not-set event_id=37000e37-f88c-4f69-a3d1-37b7b2f9e188: message=Upgrade Node perf-latency-nemesis-ubuntu-db-node-5a200ccb-1 begin
2023-07-10 19:01:32.271: (InfoEvent Severity.NORMAL) period_type=not-set event_id=f2f45f00-f446-4b7c-8e39-336d11087ca5: message=Start truncate simple tables
2023-07-10 19:03:56.799: (InfoEvent Severity.NORMAL) period_type=not-set event_id=47a33657-8a74-410f-99a8-ad955494808a: message=Finish truncate simple tables
2023-07-10 19:03:57.320: (InfoEvent Severity.NORMAL) period_type=not-set event_id=31d07e76-6250-4eb0-a5cd-81e3f2f9d810: message=Upgrading a Node
2023-07-10 19:03:57.340: (InfoEvent Severity.NORMAL) period_type=not-set event_id=3416fe53-fa00-41a0-941a-780ac30fcdd0: message=upgrade_node - removing "shim-signed" package as a workaround
2023-07-10 19:04:01.066: (InfoEvent Severity.NORMAL) period_type=not-set event_id=6a86c2ce-b7d3-4854-ad50-5ef88b6ed1d5: message=upgrade_node - ended removing "shim-signed" package as a workaround
2023-07-10 19:04:01.066: (InfoEvent Severity.NORMAL) period_type=not-set event_id=3fd4244b-cc18-4f73-af0e-d2ff8e5d2c9e: message=upgrade_node - starting to "upgrade_system" of the node perf-latency-nemesis-ubuntu-db-node-5a200ccb-1
2023-07-10 19:04:15.852: (InfoEvent Severity.NORMAL) period_type=not-set event_id=607db293-81c0-4be4-9c31-5432959e1ad2: message=upgrade_node - ended to "upgrade_system" of the node perf-latency-nemesis-ubuntu-db-node-5a200ccb-1
2023-07-10 19:04:16.510: (InfoEvent Severity.NORMAL) period_type=not-set event_id=092029e6-4050-403b-a76d-1d2894a35f15: message=upgrade_node - started to create "new_scylla_repo"
2023-07-10 19:04:17.012: (InfoEvent Severity.NORMAL) period_type=not-set event_id=3c8bf9cb-e784-4b52-951b-d865ba6b1dc2: message=upgrade_node - ended to create "new_scylla_repo"
2023-07-10 19:04:17.118: (InfoEvent Severity.NORMAL) period_type=not-set event_id=1fdc172f-e97e-485a-b21c-33952c37f4a3: message=upgrade_node - started to "backup_conf"
2023-07-10 19:04:18.114: (InfoEvent Severity.NORMAL) period_type=not-set event_id=2a3c7543-c56d-4868-9e4a-39717be83848: message=upgrade_node - ended to "backup_conf"
2023-07-10 19:04:18.114: (InfoEvent Severity.NORMAL) period_type=not-set event_id=08a60d41-203d-417e-8c25-8a414d32567c: message=upgrade_node - started to "download_scylla_repo"
2023-07-10 19:04:39.024: (InfoEvent Severity.NORMAL) period_type=not-set event_id=4c92e8a0-0a51-4de7-98c8-c1fe4b500e49: message=upgrade_node - ended to "download_scylla_repo"
2023-07-10 19:04:39.025: (InfoEvent Severity.NORMAL) period_type=not-set event_id=6de7577b-3e4d-4f94-b554-fc0b6f9fb629: message=upgrade_node - started to "drain"
2023-07-10 19:04:55.710: (InfoEvent Severity.NORMAL) period_type=not-set event_id=c56f8c4c-0d51-4ed3-9506-242363638391: message=upgrade_node - ended to "drain"
2023-07-10 19:04:55.712: (InfoEvent Severity.NORMAL) period_type=not-set event_id=f4123ab1-5324-4d03-91cf-ac62e862de3f: message=upgrade_node - started to "nodetool snapshot"
2023-07-10 19:04:56.653: (InfoEvent Severity.NORMAL) period_type=not-set event_id=1ddccfdc-b9e4-4898-918d-906ac8609739: message=upgrade_node - ended to "nodetool snapshot"
2023-07-10 19:04:56.712: (InfoEvent Severity.NORMAL) period_type=not-set event_id=26f13e8d-f701-46b8-9ec0-0437087124bc: message=upgrade_node - started to "stop_scylla_server"
2023-07-10 19:06:10.594: (InfoEvent Severity.NORMAL) period_type=not-set event_id=66270725-456a-4e29-8729-624e83bd906b: message=upgrade_node - ended to "stop_scylla_server"
2023-07-10 19:06:11.639: (InfoEvent Severity.NORMAL) period_type=not-set event_id=a24fbdc0-1bbe-423a-978f-898f809c1302: message=upgrade_node - starting to "apt-get update"
2023-07-10 19:07:45.289: (InfoEvent Severity.NORMAL) period_type=not-set event_id=9520014d-4dcd-4fca-8442-25e78894782d: message=upgrade_node - ended to "apt-get update"
2023-07-10 19:07:45.290: (InfoEvent Severity.NORMAL) period_type=not-set event_id=c439bf21-d62b-496e-a9de-7e9473ac0316: message=upgrade_node - starting to "check_reload_systemd_config"
2023-07-10 19:07:45.398: (InfoEvent Severity.NORMAL) period_type=not-set event_id=4f2bc8b4-60ee-4db2-8ef9-b0652eb8167d: message=upgrade_node - ended to "check_reload_systemd_config"
2023-07-10 19:07:45.398: (InfoEvent Severity.NORMAL) period_type=not-set event_id=07cd2896-5a5b-475a-bc48-3aaf80deadab: message=upgrade_node - starting to "run_scylla_sysconfig_setup"
2023-07-10 19:07:49.587: (InfoEvent Severity.NORMAL) period_type=not-set event_id=32e06528-9250-4c49-85d2-587354e5c13f: message=upgrade_node - ended to "run_scylla_sysconfig_setup"
2023-07-10 19:07:49.587: (InfoEvent Severity.NORMAL) period_type=not-set event_id=ce030520-3f8d-44be-84b9-a24bc040c670: message=upgrade_node - starting to "start_scylla_server"
2023-07-10 19:08:55.925: (InfoEvent Severity.NORMAL) period_type=not-set event_id=e5214a4e-dc28-465d-8d51-feef962e5be8: message=upgrade_node - ended to "start_scylla_server"
2023-07-10 19:08:55.925: (InfoEvent Severity.NORMAL) period_type=not-set event_id=f53b9564-72aa-49d5-9b5d-34e12bbd7b15: message=upgrade_node - starting to "get_db_nodes_cpu_mode"
2023-07-10 19:08:56.600: (InfoEvent Severity.NORMAL) period_type=not-set event_id=ae3a4d56-40f3-41a2-aa4f-231d41dd0666: message=upgrade_node - ended to "get_db_nodes_cpu_mode"
2023-07-10 19:08:57.265: (InfoEvent Severity.NORMAL) period_type=not-set event_id=3f43407d-622a-49ff-9673-8c19089f437c: message=upgrade_node - starting to "_update_argus_upgraded_version"
2023-07-10 19:08:57.616: (InfoEvent Severity.NORMAL) period_type=not-set event_id=b8d895a1-d37f-4f8c-8f0f-4b68e94a5196: message=upgrade_node - ended to "_update_argus_upgraded_version"
2023-07-10 19:08:57.616: (InfoEvent Severity.NORMAL) period_type=not-set event_id=4d0905b8-6539-4002-961f-cf29bea6dba8: message=upgrade_node - starting to "upgradesstables_if_command_available"
2023-07-10 19:09:00.727: (InfoEvent Severity.NORMAL) period_type=not-set event_id=a5cdcf3c-32bb-4470-afc0-dcd94b27b1bf: message=calling upgradesstables
2023-07-10 19:09:11.442: (InfoEvent Severity.NORMAL) period_type=not-set event_id=698cc9d8-6945-4775-bed6-2ee2a7512fee: message=upgrade_node - ended to "upgradesstables_if_command_available"
2023-07-10 19:09:11.560: (InfoEvent Severity.NORMAL) period_type=not-set event_id=eb9430c6-9824-4d54-99f2-48ca97941e81: message=Start read data from truncated tables
2023-07-10 19:09:11.789: (InfoEvent Severity.NORMAL) period_type=not-set event_id=4490b353-266c-479e-83bb-d38dff3f4ff1: message=Finish read data from truncated tables
2023-07-10 19:09:14.236: (InfoEvent Severity.NORMAL) period_type=not-set event_id=8507cf7b-b0ee-4560-a680-2140c61d505c: message=Upgrade Node perf-latency-nemesis-ubuntu-db-node-5a200ccb-1 ended
```

fgelcer commented:
prepare took 1 hour 42 min

fgelcer commented:
the preparation of the cluster (from `TEST_START` and until c-s prepare command started) took 35 min

fgelcer commented:
from c-s prepare end, and until c-s workload started, it took 69 minutes (checking why)

fgelcer commented:
i don't see much information actually:
```
2023-07-10 16:50:01.410: (CassandraStressEvent Severity.NORMAL) period_type=end event_id=a719a6e3-fb04-4376-9387-c0b7ec179308 duration=1h42m20s: node=Node perf-latency-nemesis-ubuntu-loader-node-5a200ccb-3 [16.170.163.237 | 10.0.3.251] (seed: False)
stress_cmd=cassandra-stress write no-warmup cl=ALL n=162500000 -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=200 -col 'size=FIXED(128) n=FIXED(8)' -pop seq=325000000..487500000
2023-07-10 16:52:06.465: (PrometheusAlertManagerEvent Severity.NORMAL) period_type=end event_id=3bae117b-9ee7-4a40-b98c-68ddba1a680c duration=1h41m59s: alert_name=cqlCLAll node=N/A start=2023-07-10T15:10:04.791Z end=2023-07-10T15:14:04.791Z description=Some queries use Consistency Level: ALL updated=2023-07-10T15:10:04.799Z state=active fingerprint=31378fe8099bfeb8 labels={'advisor': 'cqlOptimization', 'alertname': 'cqlCLAll', 'dashboard': 'cql', 'monitor': 'scylla-monitor', 'severity': '1'}
2023-07-10 16:58:52.717: (ElasticsearchEvent Severity.ERROR) period_type=one-time event_id=b4c18263-50e6-4728-a223-a0fcc3b0f9ed: doc_id=5a200ccb-e45b-462b-9a23-cc2e83ab0243_20230710_150636_786211 error=NotFoundError(404, 'document_missing_exception', '[test_stats][5a200ccb-e45b-462b-9a23-cc2e83ab0243_20230710_150636_786211]: document missing')
2023-07-10 17:28:47.206 <2023-07-10 17:28:47.000>: (DatabaseLogEvent Severity.WARNING) period_type=one-time event_id=de61c4d1-4ad7-4417-99ba-16d0fc19a9aa: type=RPC_CONNECTION regex=(^ERROR|!ERR).*rpc - client .*(connection dropped|fail to connect) line_number=6058 node=perf-latency-nemesis-ubuntu-db-node-5a200ccb-1
2023-07-10T17:28:47+00:00 perf-latency-nemesis-ubuntu-db-node-5a200ccb-1      !ERR | scylla[5403]:  [shard 3] rpc - client 162.216.150.231:64480: server connection dropped: connection is closed
2023-07-10 17:59:00.624: (CassandraStressEvent Severity.NORMAL) period_type=begin event_id=68b1b6f1-1699-403e-90b9-45ed04830f7c: node=Node perf-latency-nemesis-ubuntu-loader-node-5a200ccb-1 [16.171.47.190 | 10.0.3.230] (seed: False)
stress_cmd=cassandra-stress write no-warmup cl=QUORUM duration=2850m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate 'threads=250 fixed=20332/s' -col 'size=FIXED(128) n=FIXED(8)' -pop 'dist=gauss(1..650000000,325000000,9750000)'

```

this is between c-s prepare ending, and c-s workload starting

fgelcer commented:
the `ElasticsearchEvent` is the problem that made email report to fail to be sent.. and what Alex wrote is that we perhaps added some new "keys" to the index, and hence it failed as it hit its limit, and Alex increased that limit

fgelcer commented:
other than that, we decided that, we wait 20 min after each upgrade, to give time to cache of the node that went down to re-heat (https://github.com/scylladb/scylla-cluster-tests/pull/6182/files#diff-41fd5a7390a8abbe2d149937e3ba59997f1a83ef2fd3b49b9f3f282173347aafR879)

fgelcer commented:
@roydahan , other than that, i think it is ready to be merged

roydahan commented:
Other than it's not clear what's going on it's ready to be merged?
I don't understand above comments.
What's going on in during the 69 minutes between prepare and stress?

Also, I don't think we need to wait 20 minutes between each node upgrade, but we need to check it.

fgelcer commented:
without further change in our side, but the `upgradesstables` that now is fix, and not supposed to do anything between same sstable version, the test overall time was 7h20m ([here](https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/latency-during-upgrade/job/latency_650gb_during_upgrade/24/))
@temichus @aleksbykov , please help reviewing, so we can decide if we want to change the job or not, if we want to reduce the sleep time between the upgrades or not...

roydahan commented:
When we discussed this on google chat, we wrote:

1. reduce the nemesis_interval to 2 min (in general relevant, not only for reproducer) - this saves you 2 hours.

2. Disable the clusterHealthCheck - this saves you 30 mins.

temichus commented on performance_regression_test.py:
```diff
@@ -797,3 +798,115 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
+        self.monitoring_set = None  # pylint: disable=attribute-defined-outside-init
+
+    def get_email_data(self):  # pylint: disable=no-self-use
+        """prepare data to generate and send via email
```
please cange the comment here;

Here you force to use PerformanceRegressionTest 
email tade generation  mechanism instead of UpgradeTest

or maybe even better to call PerformanceRegressionTest.get_email_data function directly here
```
def get_email_data(self):
   return PerformanceRegressionTest.get_email_data(self)
``

fgelcer replied:
because of the multiple inheritance, we got into a situation where `get_email_data()` is not implemented in `PerformanceRegressionTest`, but only on `UpgradeTest` leading to errors...
so, in this case, we need to do the same as `PerformanceRegressionTest` does (from its base class) that is `return {}`

temichus replied:
just look at the code :
```
 Required field:
        - email_template: path to file with html template
        - email_subject: subject of email
        have to be implemented in child class.
```
It looks like an instruction that we need to implement this method

but as you mentioned for propperf work it must call the same method as in  PerformanceRegressionTest 

so we can call the method PerformanceRegressionTest directly to skip MRO  
```
def get_email_data(self):
   return PerformanceRegressionTest.get_email_data(self)
```

fgelcer commented:
@roydahan ,with current changes the test took 5 hours 52 min
we could make the "steady state" and the "post upgrade steady state" slight smaller, and save another 20 - 40 min.. but is that really required?

fgelcer commented:
> > @roydahan ,with current changes the test took 5 hours 52 min we could make the "steady state" and the "post upgrade steady state" slight smaller, and save another 20 - 40 min.. but is that really required?
> 
> looks like the upgrade itself took 40min according. + 60min for "steady state". Cant understand from logs what else happening in the test for 4h. @fgelcer Could you please clarify?

steady state is 30 minutes (same for post upgrade steady state)

i will tomorrow open logs and try to get the times of each action and describe them here

temichus commented:
> > > @roydahan ,with current changes the test took 5 hours 52 min we could make the "steady state" and the "post upgrade steady state" slight smaller, and save another 20 - 40 min.. but is that really required?
> > 
> > 
> > looks like the upgrade itself took 40min according. + 60min for "steady state". Cant understand from logs what else happening in the test for 4h. @fgelcer Could you please clarify?
> 
> steady state is 30 minutes (same for post upgrade steady state)
> 
> i will tomorrow open logs and try to get the times of each action and describe them here

steady state is 30 minutes (same for post upgrade steady state) - yes 30+30 =60

fgelcer commented:
> When we discussed this on google chat, we wrote:
> 
> 1. reduce the nemesis_interval to 2 min (in general relevant, not only for reproducer) - this saves you 2 hours.
> 2. Disable the clusterHealthCheck - this saves you 30 mins.

@roydahan , both addressed in this run --> https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/latency-during-upgrade/job/latency_650gb_during_upgrade/28/

roydahan commented on performance_regression_test.py:
```diff
@@ -797,3 +798,115 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
+
+    @latency_calculator_decorator
+    def steady_state_latency(self):  # pylint: disable=no-self-use
+        sleep_time = 1800
```
What is this sleep time?
Why it's 1800 and why it's not controlled from outside parameter?

fgelcer replied:
this is THE steady state
this is 1800, because we wanted to reduce the default one (default, based on nemesis interval we have for the config file) that is 3600

if changing this one to the nemesis interval back again, it will be twice as the current value (but controllable from test config file)... so, should i also reduce the `nemesis_interval` (it will affect the nemesis during operations too)

roydahan commented on performance_regression_test.py:
```diff
@@ -797,3 +798,115 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
+
+    @latency_calculator_decorator
+    def post_upgrades_steady_state(self):
+        sleep_time = 1800
```
same

fgelcer replied:
ditto

fgelcer commented:
> > When we discussed this on google chat, we wrote:
> > 
> > 1. reduce the nemesis_interval to 2 min (in general relevant, not only for reproducer) - this saves you 2 hours.
> > 2. Disable the clusterHealthCheck - this saves you 30 mins.
> 
> @roydahan , both addressed in this run --> https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/latency-during-upgrade/job/latency_650gb_during_upgrade/28/

@roydahan , write workload failed... results were sent for read and mixed:

```
[aws][i3en.2xlarge] Performance Regression Compare Results (latency during upgrades 650gb) - test_latency_read_with_upgrade - 5.4.0~dev-0.20230608.ce6a1ca13b59 - 2023-09-11 11:01:00
```

```
[aws][i3en.2xlarge] Performance Regression Compare Results (latency during upgrades 650gb) - test_latency_mixed_with_upgrade - 5.4.0~dev-0.20230608.ce6a1ca13b59 - 2023-09-11 11:25:19
```

these emails were sent to `qa-perf-team`, so you can check the results yourself, so we can define what is missing, so we can close this task, that IMHO is ready to be merged, and if follow up adjustments will be needed, we can simply change later

fgelcer commented:
ah, didn't see last comments... will address them tomorrow/today

roydahan commented on upgrade_test.py:
```diff
@@ -161,9 +161,12 @@ def validate_truncated_entries_for_table(self, session, system_truncated=False):
                                 msg='Expected truncated entry in the system.local table, but it\'s not found')
 
     @truncate_entries
+    def upgrade_node(self, node, upgrade_sstables=True):
```
I recall there was a comment here about the "truncate_entries" that it requires the test to have some info in that table in order for the truncate not to fail.
What was the outcome of it?

fgelcer replied:
i moved the call in this function to a common "private" function, that is what i call too, without the `truncate_entries` decorator

roydahan commented:
> > > When we discussed this on google chat, we wrote:
> > > 
> > > 1. reduce the nemesis_interval to 2 min (in general relevant, not only for reproducer) - this saves you 2 hours.
> > > 2. Disable the clusterHealthCheck - this saves you 30 mins.
> > 
> > 
> > @roydahan , both addressed in this run --> https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/latency-during-upgrade/job/latency_650gb_during_upgrade/28/
> 
> @roydahan , write workload failed... results were sent for read and mixed:
> 
> ```
> [aws][i3en.2xlarge] Performance Regression Compare Results (latency during upgrades 650gb) - test_latency_read_with_upgrade - 5.4.0~dev-0.20230608.ce6a1ca13b59 - 2023-09-11 11:01:00
> ```
> 
> ```
> [aws][i3en.2xlarge] Performance Regression Compare Results (latency during upgrades 650gb) - test_latency_mixed_with_upgrade - 5.4.0~dev-0.20230608.ce6a1ca13b59 - 2023-09-11 11:25:19
> ```
> 
> these emails were sent to `qa-perf-team`, so you can check the results yourself, so we can define what is missing, so we can close this task, that IMHO is ready to be merged, and if follow up adjustments will be needed, we can simply change later

1. Why did the write workload failed? Is it related to the test? I don't want to push the problem to later for someone else to handle.
2. I saw the emails and saw that each cycle still takes apx 1h15m to upgrade each node, why is that?
What's the breakdown of the operations and what steps take the most?
(Not you have info events, it should be east to show the breakdown).

fgelcer commented:
> > > > When we discussed this on google chat, we wrote:
> > > > 
> > > > 1. reduce the nemesis_interval to 2 min (in general relevant, not only for reproducer) - this saves you 2 hours.
> > > > 2. Disable the clusterHealthCheck - this saves you 30 mins.
> > > 
> > > 
> > > @roydahan , both addressed in this run --> https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/latency-during-upgrade/job/latency_650gb_during_upgrade/28/
> > 
> > 
> > @roydahan , write workload failed... results were sent for read and mixed:
> > ```
> > [aws][i3en.2xlarge] Performance Regression Compare Results (latency during upgrades 650gb) - test_latency_read_with_upgrade - 5.4.0~dev-0.20230608.ce6a1ca13b59 - 2023-09-11 11:01:00
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > [aws][i3en.2xlarge] Performance Regression Compare Results (latency during upgrades 650gb) - test_latency_mixed_with_upgrade - 5.4.0~dev-0.20230608.ce6a1ca13b59 - 2023-09-11 11:25:19
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > these emails were sent to `qa-perf-team`, so you can check the results yourself, so we can define what is missing, so we can close this task, that IMHO is ready to be merged, and if follow up adjustments will be needed, we can simply change later
> 
> 1. Why did the write workload failed? Is it related to the test? I don't want to push the problem to later for someone else to handle.

i'm still didn't investigate it further, but it failed during prepare stage (this is exactly same workload as we use for the latency during operations, so nothing special in this test that could have broken it

> 2. I saw the emails and saw that each cycle still takes apx 1h15m to upgrade each node, why is that?

i will list the breakdown of the events we printed

>    What's the breakdown of the operations and what steps take the most?
>    (Not you have info events, it should be east to show the breakdown).

BTW, i reduced the `nemesis_interval` for both this, and latency during operations... also, i have removed the truncate (did not run with it yet).. so let's see the breakdown and see how to make progress

fgelcer commented:
here is the breakdown:
```
2023-09-11 11:59:04.987: (InfoEvent Severity.NORMAL) period_type=not-set event_id=c03936fb-310d-4dae-8595-180a45b12e57: message=Upgrade Node perf-latency-nemesis-ubuntu-db-node-3e7eb6a5-1 begin
2023-09-11 11:59:07.487: (InfoEvent Severity.NORMAL) period_type=not-set event_id=f567d6d3-133c-423d-9f6f-08f1e95d7497: message=Start truncate simple tables
2023-09-11 12:00:16.576: (InfoEvent Severity.NORMAL) period_type=not-set event_id=c6df770d-e4eb-4e71-b9d9-a8c37616896d: message=Finish truncate simple tables
2023-09-11 12:00:17.570: (InfoEvent Severity.NORMAL) period_type=not-set event_id=72150810-edbc-4d6b-a971-78f2b18cb51f: message=Upgrading a Node
2023-09-11 12:00:17.571: (InfoEvent Severity.NORMAL) period_type=not-set event_id=87047f09-9a7e-45a0-b49c-f8855c5284c7: message=upgrade_node - removing "shim-signed" package as a workaround
2023-09-11 12:00:20.197: (InfoEvent Severity.NORMAL) period_type=not-set event_id=871b85f6-73bc-4a2e-8cb5-c0357ef932b6: message=upgrade_node - ended removing "shim-signed" package as a workaround
2023-09-11 12:00:20.198: (InfoEvent Severity.NORMAL) period_type=not-set event_id=da9f6c70-89d2-4887-a330-4741d0ba42dc: message=upgrade_node - starting to "upgrade_system" of the node perf-latency-nemesis-ubuntu-db-node-3e7eb6a5-1
2023-09-11 12:00:42.539: (InfoEvent Severity.NORMAL) period_type=not-set event_id=77060676-90d7-4aa8-925d-9125d8177739: message=upgrade_node - ended to "upgrade_system" of the node perf-latency-nemesis-ubuntu-db-node-3e7eb6a5-1
2023-09-11 12:00:43.141: (InfoEvent Severity.NORMAL) period_type=not-set event_id=827a20a0-216d-4e4c-9093-92be6a8e16e7: message=upgrade_node - started to create "new_scylla_repo"
2023-09-11 12:00:44.469: (InfoEvent Severity.NORMAL) period_type=not-set event_id=0dc9ab40-4a80-4f9d-8d41-759a72e73603: message=upgrade_node - ended to create "new_scylla_repo"
2023-09-11 12:00:44.479: (InfoEvent Severity.NORMAL) period_type=not-set event_id=32df0b52-7481-47f3-9c49-015fa425637a: message=upgrade_node - started to "backup_conf"
2023-09-11 12:00:46.045: (InfoEvent Severity.NORMAL) period_type=not-set event_id=0577f445-e573-40f9-9658-bea000ed0391: message=upgrade_node - ended to "backup_conf"
2023-09-11 12:00:46.441: (InfoEvent Severity.NORMAL) period_type=not-set event_id=33f04b4f-f742-41b2-9067-019afca5c6fa: message=upgrade_node - started to "download_scylla_repo"
2023-09-11 12:01:07.175: (InfoEvent Severity.NORMAL) period_type=not-set event_id=45bd91fe-9569-4444-b241-b03a0423c62c: message=upgrade_node - ended to "download_scylla_repo"
2023-09-11 12:01:07.190: (InfoEvent Severity.NORMAL) period_type=not-set event_id=7eaaf61e-156c-45a5-8c81-2834afd0211d: message=upgrade_node - started to "drain"
2023-09-11 12:01:18.447: (InfoEvent Severity.NORMAL) period_type=not-set event_id=845b6927-f168-4c3b-a622-68ac5a041eb3: message=upgrade_node - ended to "drain"
2023-09-11 12:01:18.447: (InfoEvent Severity.NORMAL) period_type=not-set event_id=21e0f9b2-2cc5-4961-adc8-c33c9e1a1ea8: message=upgrade_node - started to "nodetool snapshot"
2023-09-11 12:01:19.821: (InfoEvent Severity.NORMAL) period_type=not-set event_id=08a39be1-a130-470e-a72e-ac11b0190263: message=upgrade_node - ended to "nodetool snapshot"
2023-09-11 12:01:19.870: (InfoEvent Severity.NORMAL) period_type=not-set event_id=89ccd1e4-dfdf-489b-994d-16b107502d9e: message=upgrade_node - started to "stop_scylla_server"
2023-09-11 12:02:38.704: (InfoEvent Severity.NORMAL) period_type=not-set event_id=1c705100-f8d3-4be4-969e-dc602c6a41e7: message=upgrade_node - ended to "stop_scylla_server"
2023-09-11 12:02:39.721: (InfoEvent Severity.NORMAL) period_type=not-set event_id=b27b7de9-e4c5-4606-88d1-b480efd9f947: message=upgrade_node - starting to "apt-get update"
2023-09-11 12:04:44.253: (InfoEvent Severity.NORMAL) period_type=not-set event_id=4f572b73-d59f-461f-b952-124b042d9e98: message=upgrade_node - ended to "apt-get update"
2023-09-11 12:04:44.572: (InfoEvent Severity.NORMAL) period_type=not-set event_id=3f434462-dd30-49b5-bf0d-bb77f6c5113b: message=upgrade_node - starting to "check_reload_systemd_config"
2023-09-11 12:04:46.377: (InfoEvent Severity.NORMAL) period_type=not-set event_id=aadc2152-cae8-4dd1-b619-0b534cc805d6: message=upgrade_node - ended to "check_reload_systemd_config"
2023-09-11 12:04:46.499: (InfoEvent Severity.NORMAL) period_type=not-set event_id=0df1e4ad-5a88-44f4-83bd-0d78c7886a9b: message=upgrade_node - starting to "run_scylla_sysconfig_setup"
2023-09-11 12:04:50.637: (InfoEvent Severity.NORMAL) period_type=not-set event_id=ef9b7eea-3c71-48eb-8af8-08488fb30143: message=upgrade_node - ended to "run_scylla_sysconfig_setup"
2023-09-11 12:04:50.638: (InfoEvent Severity.NORMAL) period_type=not-set event_id=ee795aef-ad35-44fe-9dd4-571a33c04fea: message=upgrade_node - starting to "start_scylla_server"
2023-09-11 12:05:56.439: (InfoEvent Severity.NORMAL) period_type=not-set event_id=cfd81204-81c7-4e20-b802-0a6258829a14: message=upgrade_node - ended to "start_scylla_server"
2023-09-11 12:05:56.586: (InfoEvent Severity.NORMAL) period_type=not-set event_id=e6d12219-ec1e-4b20-ae8b-1872f28383ad: message=upgrade_node - starting to "get_db_nodes_cpu_mode"
2023-09-11 12:06:41.933: (InfoEvent Severity.NORMAL) period_type=not-set event_id=0faea330-3ca6-4386-aca3-f849787a64d6: message=upgrade_node - ended to "get_db_nodes_cpu_mode"
2023-09-11 12:06:42.463: (InfoEvent Severity.NORMAL) period_type=not-set event_id=0640aac2-1fbb-4ac0-884b-ebc6396f434a: message=upgrade_node - starting to "_update_argus_upgraded_version"
2023-09-11 12:06:43.580: (InfoEvent Severity.NORMAL) period_type=not-set event_id=725dcf7f-3470-454d-99cd-f652b891913b: message=upgrade_node - ended to "_update_argus_upgraded_version"
2023-09-11 12:06:43.581: (InfoEvent Severity.NORMAL) period_type=not-set event_id=f39ee494-818d-423f-922d-2196d444fa6a: message=upgrade_node - starting to "upgradesstables_if_command_available"
2023-09-11 12:06:46.220: (InfoEvent Severity.NORMAL) period_type=not-set event_id=59376b70-2323-411a-9ea3-a0c69f0cfef9: message=calling upgradesstables
2023-09-11 12:06:51.019: (InfoEvent Severity.NORMAL) period_type=not-set event_id=c0710672-c580-4176-ac33-e7d3ad28ace1: message=upgrade_node - ended to "upgradesstables_if_command_available"
2023-09-11 12:06:51.069: (InfoEvent Severity.NORMAL) period_type=not-set event_id=eeaf8252-468d-426d-b672-643bae326f41: message=Start read data from truncated tables
2023-09-11 12:06:58.831: (InfoEvent Severity.NORMAL) period_type=not-set event_id=10613f8e-c822-4a2e-9581-32968141f512: message=Finish read data from truncated tables
2023-09-11 13:31:17.614: (InfoEvent Severity.NORMAL) period_type=not-set event_id=faaac6e2-bff1-43cd-ae4c-7d81d3305684: message=Upgrade Node perf-latency-nemesis-ubuntu-db-node-3e7eb6a5-1 ended
```

so, most of the time was between `2023-09-11 12:06:58.831: (InfoEvent Severity.NORMAL) period_type=not-set event_id=10613f8e-c822-4a2e-9581-32968141f512: message=Finish read data from truncated tables` and `2023-09-11 13:31:17.614: (InfoEvent Severity.NORMAL) period_type=not-set event_id=faaac6e2-bff1-43cd-ae4c-7d81d3305684: message=Upgrade Node perf-latency-nemesis-ubuntu-db-node-3e7eb6a5-1 ended`

but it doesn't make sense..
i'm trying to find why

fgelcer commented:
part of the time was wasted inserting data into `truncate_table`.. but between these insert commands, we have many:
```
< t:2023-09-11 12:19:04,267 f:db_log_reader.py l:114  c:sdcm.db_log_reader   p:DEBUG > 2023-09-11T12:19:04+00:00 perf-latency-nemesis-ubuntu-db-node-3e7eb6a5-1  !WARNING | scylla[101767]:  [shard 0] lsa-timing - reclaim took 12000 us, trying to release 4.201 MiB preemptibly, reserve: {goal: 1, max: 30}, at 0x5a11b6e 0x5a12120 0x5a12408 0x1e9b8ad 0x1e99a9c 0x1eb9162 0x12921fa 0x5645be4 0x5646e67 0x56461a9 0x55ea0c5 0x55e9238 0x121a493 0x121c020 0x12188b7 /opt/scylladb/libreloc/libc.so.6+0x2750f /opt/scylladb/libreloc/libc.so.6+0x275c8 0x1216464
< t:2023-09-11 12:19:04,268 f:db_log_reader.py l:114  c:sdcm.db_log_reader   p:DEBUG > --------
< t:2023-09-11 12:19:04,268 f:db_log_reader.py l:114  c:sdcm.db_log_reader   p:DEBUG > seastar::internal::coroutine_traits_base<void>::promise_type
< t:2023-09-11 12:19:04,268 f:db_log_reader.py l:114  c:sdcm.db_log_reader   p:DEBUG > 2023-09-11T12:19:04+00:00 perf-latency-nemesis-ubuntu-db-node-3e7eb6a5-1     !INFO | scylla[101767]:  [shard 0] lsa-timing - - reclamation rate = 312.500 MiB/s
< t:2023-09-11 12:19:04,269 f:db_log_reader.py l:114  c:sdcm.db_log_reader   p:DEBUG > 2023-09-11T12:19:04+00:00 perf-latency-nemesis-ubuntu-db-node-3e7eb6a5-1     !INFO | scylla[101767]:  [shard 0] lsa-timing - - compacted segments: 3
< t:2023-09-11 12:19:04,269 f:db_log_reader.py l:114  c:sdcm.db_log_reader   p:DEBUG > 2023-09-11T12:19:04+00:00 perf-latency-nemesis-ubuntu-db-node-3e7eb6a5-1     !INFO | scylla[101767]:  [shard 0] lsa-timing - - compacted memory: 0.022 MiB
```

fgelcer commented:
and we have many:
```
< t:2023-09-11 12:19:37,381 f:load_info_store.py l:217  c:sdcm.utils.adaptive_timeouts.load_info_store p:DEBUG > Storing adaptive timeout info: 0dcee61c-aa04-41dd-a93b-4d1da7ca3f40={'node_name': 'perf-latency-nemesis-ubuntu-db-node-3e7eb6a5-1', 'cpu_load_5': 7.31, 'shards_count': 7, 'read_bandwidth_mb': 1272.0, 'write_bandwidth_mb': 636.0, 'read_iops': 168960, 'write_iops': 133938, 'node_data_size_mb': 734863, 'scylla_version': '5.2.7-0.20230821.e0ebc95025d1 with build-id 0dff18311edc6ec24a66cfcfa280b6a150ab9fc5', 'query': 'INSERT', 'test_id': '3e7eb6a5-f24e-469c-9e3a-3e51bb6a5eb6', 'end_time': datetime.datetime(2023, 9, 11, 12, 19, 37, 381245), 'operation': 'QUERY', 'duration': 0.1922919750213623, 'timeout': 60, 'timeout_occurred': False}
< t:2023-09-11 12:19:37,476 f:base.py         l:274  c:elasticsearch        p:INFO  > PUT https://746f0ad652a3447d83b1572f657c67cb.us-east-1.aws.found.io:9243/sct-adaptive-timeouts/_create/0dcee61c-aa04-41dd-a93b-4d1da7ca3f40 [status:201 request:0.090s]
< t:2023-09-11 12:19:37,476 f:base.py         l:277  c:elasticsearch        p:DEBUG > > {"node_name":"perf-latency-nemesis-ubuntu-db-node-3e7eb6a5-1","cpu_load_5":7.31,"shards_count":7,"read_bandwidth_mb":1272.0,"write_bandwidth_mb":636.0,"read_iops":168960,"write_iops":133938,"node_data_size_mb":734863,"scylla_version":"5.2.7-0.20230821.e0ebc95025d1 with build-id 0dff18311edc6ec24a66cfcfa280b6a150ab9fc5","query":"INSERT","test_id":"3e7eb6a5-f24e-469c-9e3a-3e51bb6a5eb6","end_time":"2023-09-11T12:19:37.381245","operation":"QUERY","duration":0.1922919750213623,"timeout":60,"timeout_occurred":false}
< t:2023-09-11 12:19:37,477 f:base.py         l:278  c:elasticsearch        p:DEBUG > < {"_index":"sct-adaptive-timeouts","_type":"_doc","_id":"0dcee61c-aa04-41dd-a93b-4d1da7ca3f40","_version":1,"result":"created","_shards":{"total":2,"successful":2,"failed":0},"_seq_no":154870,"_primary_term":10}
```

fgelcer commented on test-cases/performance/perf-regression-latency-650gb-with-nemesis.yaml:
```diff
@@ -18,7 +18,7 @@ instance_type_monitor: 't3.large'
 
 nemesis_class_name: 'NemesisSequence'
-nemesis_interval: 60
+nemesis_interval: 30
```
@roydahan @temichus @aleksbykov , changed it here, and it may affect somewhat latency during operations too, FYI

roydahan replied:
I saw it, but how shall it affect it?
(And if it does, we need to remove the change)

fgelcer replied:
it will reduce the steady state... so my idea is now (next) to check the monitor of latest run, and confirm how long it does take for the "memory/cache" take to stabilize, during the steady state...
we have some time we wait beforehand, to heat the cache, so it should not affect at all, but need to be sure

temichus replied:
@fgelcer, did you check your idea?

fgelcer commented:
@roydahan @temichus @aleksbykov , latest run finished with success, and duration was 4 hours..
checking email, and will paste here the breakdown of info events, but surely removing the truncate improved a lot the times

fgelcer commented:
for mixed, operation (upgrade) times were:
6 min
5.5 min
5 min

fgelcer commented:
```
message=Upgrade Node perf-latency-nemesis-ubuntu-db-node-98d280e4-1 begin
message=Upgrading a Node
message=upgrade_node - removing "shim-signed" package as a workaround
message=upgrade_node - ended removing "shim-signed" package as a workaround
message=upgrade_node - starting to "upgrade_system" of the node perf-latency-nemesis-ubuntu-db-node-98d280e4-1
message=upgrade_node - ended to "upgrade_system" of the node perf-latency-nemesis-ubuntu-db-node-98d280e4-1
message=upgrade_node - started to create "new_scylla_repo"
message=upgrade_node - ended to create "new_scylla_repo"
message=upgrade_node - started to "backup_conf"
message=upgrade_node - ended to "backup_conf"
message=upgrade_node - started to "download_scylla_repo"
message=upgrade_node - ended to "download_scylla_repo"
message=upgrade_node - started to "drain"
message=upgrade_node - ended to "drain"
message=upgrade_node - started to "nodetool snapshot"
message=upgrade_node - ended to "nodetool snapshot"
message=upgrade_node - started to "stop_scylla_server"
message=upgrade_node - ended to "stop_scylla_server"
message=upgrade_node - starting to "apt-get update"
message=upgrade_node - ended to "apt-get update"
message=upgrade_node - starting to "check_reload_systemd_config"
message=upgrade_node - ended to "check_reload_systemd_config"
message=upgrade_node - starting to "run_scylla_sysconfig_setup"
message=upgrade_node - ended to "run_scylla_sysconfig_setup"
message=upgrade_node - starting to "start_scylla_server"
message=upgrade_node - ended to "start_scylla_server"
message=upgrade_node - starting to "get_db_nodes_cpu_mode"
message=upgrade_node - ended to "get_db_nodes_cpu_mode"
message=upgrade_node - starting to "_update_argus_upgraded_version"
message=upgrade_node - ended to "_update_argus_upgraded_version"
message=upgrade_node - starting to "upgradesstables_if_command_available"
message=calling upgradesstables
message=upgrade_node - ended to "upgradesstables_if_command_available"
message=Upgrade Node perf-latency-nemesis-ubuntu-db-node-98d280e4-1 ended
```

fgelcer commented:
ping @temichus @aleksbykov @roydahan

roydahan commented:
Now results make much more sense.

aleksbykov commented on performance_regression_test.py:
```diff
@@ -797,3 +798,109 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
+class PerformanceRegressionUpgradeTest(PerformanceRegressionTest, UpgradeTest):  # pylint: disable=too-many-ancestors
+    def __int__(self, *args):
+        super().__init__(*args)
+        self.cluster = None  # pylint: disable=attribute-defined-outside-init
+        self.tester = None  # pylint: disable=attribute-defined-outside-init
+        self.monitoring_set = None  # pylint: disable=attribute-defined-outside-init
```
Does still pylint disable comments required?

fgelcer replied:
i believe so.. i can try to remove them locally, and run pre-commit, but they are defined outside of `__init__()` method, so the pylint disable should be still required

vponomaryov replied:
Actually you have typo here. The letter `i` is lost in the `__int__` method name.
So, lines 806-808 make no any effect to the logic.
If the code did work for you, then the definition of these attributes was not really needed...

aleksbykov reviewed (approved):
LGTM

temichus reviewed (commented):
still disagree with get_email_data
```
 Required field:
        - email_template: path to file with html template
        - email_subject: subject of email
 have to be implemented in child class.
 ```
It looks like an instruction that we need to implement this method

but as you mentioned for propperf work it must call the same method as in PerformanceRegressionTest

so we can call the method PerformanceRegressionTest directly to skip MRO
```
def get_email_data(self):
   return PerformanceRegressionTest.get_email_data(self)
```

temichus commented:
> > still disagree with get_email_data
> > ```
> >  Required field:
> >         - email_template: path to file with html template
> >         - email_subject: subject of email
> >  have to be implemented in child class.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It looks like an instruction that we need to implement this method
> > but as you mentioned for propperf work it must call the same method as in PerformanceRegressionTest
> > so we can call the method PerformanceRegressionTest directly to skip MRO
> > ```
> > def get_email_data(self):
> >    return PerformanceRegressionTest.get_email_data(self)
> > ```
> 
> Again, this is an instruction AFAIU from a test that is not related to perf regression jobs, and perf regression jobs have a special way to deal with email data/send emails... so, this is something that you @temichus show me yourself how to fix, when at first, i couldn't send the email reports

Please remove comment in this case and just leave it as:

```
def get_email_data(self):
     return {}
```

or more propper(but result will be the same)

```
def get_email_data(self):
    return PerformanceRegressionTest.get_email_data(self)
```

fgelcer commented:
```
def get_email_data(self):
    return PerformanceRegressionTest.get_email_data(self)
```

this above will fail, as `PerformanceRegressionTest` doesn't have a method `get_email_data`


removed the docstring, and re-submitting

temichus commented:
> ```
> def get_email_data(self):
>     return PerformanceRegressionTest.get_email_data(self)
> ```
> 
> this above will fail, as `PerformanceRegressionTest` doesn't have a method `get_email_data`
> 
> removed the docstring, and re-submitting

it has, in ClusterTester https://github.com/scylladb/scylla-cluster-tests/blob/e66e395388542c7896f614caf9b8d124f4344860/sdcm/tester.py#L3536
```
>>> PerformanceRegressionTest.mro()
[<class 'performance_regression_test.PerformanceRegressionTest'>, <class 'sdcm.tester.ClusterTester'>, <class 'sdcm.db_stats.TestStatsMixin'>, <class 'sdcm.db_stats.Stats'>, <class 'unittest.case.TestCase'>, <class 'object'>]
```


but ```return {}``` also ok for me

fgelcer commented:
@roydahan @temichus @aleksbykov , addressed all comments.. ready to be merged

fgelcer commented:
@roydahan , if you merge it now, we maybe can get results for a run until the meeting at 12:00

fgelcer commented:
@roydahan , still waits for you

fgelcer commented:
@fruch , FYI

vponomaryov commented on performance_regression_test.py:
```diff
@@ -797,3 +798,100 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
+
+    @latency_calculator_decorator(legend="Upgrade Node")
+    def upgrade_node(self, node):  # pylint: disable=arguments-differ
+        InfoEvent(message='Upgrade Node %s begin' % node.name).publish()
```
For this kind of functionality we have `log_run_info` decorator.
So, I think it must be reused here.

fgelcer replied:
removing this code block, as if this typo was already in there beforehand, it really is not necessary...

fgelcer replied:
it won't give us the node name, would it?
because the message is declared before we have a "node", so we won't really know in which node the upgrade started

vponomaryov replied:
No, node name won't be added.

fgelcer replied:
so not doing it (for not having the node's name), but also because with using `log_run_info` we won't see it annotated in Grafana

vponomaryov commented on performance_regression_test.py:
```diff
@@ -797,3 +798,100 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
+            self.loaders.kill_stress_thread()
+
+    @latency_calculator_decorator
+    def steady_state_latency(self):  # pylint: disable=no-self-use
```
The same here. Use `log_run_info`

fgelcer replied:
won't do it, as with using `log_run_info` we won't see it annotated in Grafana

vponomaryov commented on performance_regression_test.py:
```diff
@@ -797,3 +798,100 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
+        InfoEvent(message='Ended Steady State calculation. Took %ss' % sleep_time).publish()
+
+    @latency_calculator_decorator
+    def post_upgrades_steady_state(self):
```
log_run_info

fgelcer replied:
won't do it, as with using `log_run_info` we won't see it annotated in Grafana

fgelcer commented:
> Actually you have typo here. The letter i is lost in the __int__ method name.
> So, lines 806-808 make no any effect to the logic.
> If the code did work for you, then the definition of these attributes was not really needed...


so, removed the function with typo, and apparently it worked... waiting for the test to end, to confirm all the calculations will work correclty, but the other changes requested won't be done, because they are not doing what the info events are doing for this test, that is adding annotations to Grafana... @vponomaryov , if you still have concerns, comment, if not, re-review and approve

fgelcer commented:
@roydahan

vponomaryov reviewed (approved):
LGTM

roydahan commented:
Now need to re review, since I'm not sure what you changed.

fgelcer commented:
> > Now need to re review, since I'm not sure what you changed.
> 
> the only thing that changed, is that i removed `def __int__` because is wasn't being used (it has a typo, instead of `__init__`) and worked either way... so i removed it, and re-run the test, so it passed, and sent correctly the report by email... everything else is the same as your last review

@roydahan , you can see the difference is from https://github.com/scylladb/scylla-cluster-tests/commit/edeb927b2389276694bc82e58e52f62e792f5a8f to https://github.com/scylladb/scylla-cluster-tests/commit/b437f1645199ebd204730cbaa3e77d8f79a385fa

fgelcer commented:
> Doesn't backport cleanly.

backport PR is #6637
